src/coins.cpp

```diff
@@ -280,10 +282,21 @@ uint256 CCoinsViewCache::GetBestBlock() const {
 }
 
 
-uint256 CCoinsViewCache::GetBestAnchor() const {
-    if (hashSproutAnchor.IsNull())
-        hashSproutAnchor = base->GetBestAnchor();
-    return hashSproutAnchor;
+uint256 CCoinsViewCache::GetBestAnchor(ShieldedType type) const {
+    switch (type) {
+        case SPROUT:
+            if (hashSproutAnchor.IsNull())
+                hashSproutAnchor = base->GetBestAnchor(type);
+            return hashSproutAnchor;
+            break;
+        case SAPLING:
+            if (hashSaplingAnchor.IsNull())
+                hashSaplingAnchor = base->GetBestAnchor(type);
+            return hashSaplingAnchor;
+            break;
+        default:
+            throw std::runtime_error("Unknown shielded type " + type);
```

Comment:
This is going to cause warnings (and thus CI failures) in clang/MacOS. Either convert `type` to a string, use a string format function, or remove it.

Reply:
This is in the other code handling nullifiers so let's deal with this in another PR.

Reply:
We already removed the nullifiers cases in #3227.